parser.y

```diff
@@ -2332,22 +2334,24 @@ CreateIndexStmt:
 			}
 		}
 		$$ = &ast.CreateIndexStmt{
-			Unique:        $2.(bool),
 			IfNotExists:   $4.(bool),
 			IndexName:     $5,
 			Table:         $8.(*ast.TableName),
 			IndexColNames: $10.([]*ast.IndexColName),
 			IndexOption:   indexOption,
+			KeyType:       $2.(ast.IndexKeyType),
+			Algorithm:     $14.(*ast.AlterAlgorithm),
 		}
 	}
 
-CreateIndexStmtUnique:
+IndexAlgorithmOpt:
 	{
-		$$ = false
+		$$ = (*ast.AlterAlgorithm)(nil)
```

Comment:
```suggestion
		$$ = nil
```

Reply:
I'm afraid that we have to do this. If `nil` has not been converted, there will be a panic:
```
... Panic: interface conversion: interface {} is nil, not *ast.AlterAlgorithm (PC=0x456A94)
```